### PR TITLE
Enabling editing an advertisement

### LINKED ISSUE #1157: Enabling editing an advertisement
**Is your feature request related to a problem? Please describe.**
We have ensured CRUD operations throughout the talawa( wherever necessary). We require an edit ad modal in the Advertisement section on all the created ads.

**Describe the solution you'd like**
An editing ad modal on all created ads, ensuring fields similar to create ads modal.

**Additional context**
NA

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Vaishali054**
**What kind of change does this PR introduce?**

Enabling editing an advertisement

**Issue Number:**

Closes #1157  

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa-admin/assets/100770875/4eed8fb9-f9f7-4be5-ab65-6a48d9ab89a6

Advertisement Type Dropdown

Screenshot 2024-01-15 143054

Date selector

Screenshot 2024-01-15 143124

**If relevant, did you update the documentation?**

NA

**Summary**

Adding edit advertisement modal

**Does this PR introduce a breaking change?**

No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by palisadoes** [comment]
1. Please fix the introspection tests. 
1. Update any items that may not have been a part of this PR. We merged a previous one without checking sufficiently

**Comment by beingnoble03** [comment]
@Vaishali054 the dropdown isn't visible in the added video recording. Can you please attach a screenshot of the dropdown or update the video?

**Comment by Vaishali054** [comment]
@beingnoble03 added the snapshots of the dropdowns. Please check.

**Comment by beingnoble03** [comment]
Also fix the failing tests

**Comment by Vaishali054** [comment]
It is due to the bug found in one of the workflows. The issue #1429 will fix this. I will add a commit to trigger re-run of the workflows once this issue is resolved.

File: src/components/Advertisements/Advertisements.test.tsx [hunk 1]
```diff
@@ -163,57 +163,6 @@ const ORGANIZATIONS_LIST_MOCK = {
 };
 
 describe('Testing Advertisement Component', () => {
-  test('Testing if the deletion works', async () => {
```
Comment (beingnoble03) [inline src/components/Advertisements/Advertisements.test.tsx:166]: Why is this test removed? Is it moved to `AdvertisementEntry.test.tsx`?

File: src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.tsx [hunk 2]
```diff
@@ -29,8 +31,8 @@ function advertisementEntry({
 }: InterfaceAddOnEntryProps): JSX.Element {
   const { t } = useTranslation('translation', { keyPrefix: 'advertisement' });
   const [buttonLoading, setButtonLoading] = useState(false);
+  const [dropdown, setDropdown] = useState(false);
   const [showDeleteModal, setShowDeleteModal] = useState(false);
-
```
Comment (beingnoble03) [inline src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.tsx:33]: Don't delete this line. It was added to separate state variables from `useMutation` variables.

File: src/components/Advertisements/core/AdvertisementRegister/AdvertisementRegister.tsx [hunk 3]
```diff
@@ -77,20 +123,88 @@ function advertisementRegister({
...
     <>
-      <Button
-        className={styles.modalbtn}
-        variant="primary"
-        onClick={handleShow}
-      >
-        <i className="fa fa-plus"></i>
-        {t('addNew')}
-      </Button>
+      {formStatus === 'register' ? ( //If register show register button else show edit button
```
Comment (beingnoble03) [inline src/components/Advertisements/core/AdvertisementRegister/AdvertisementRegister.tsx:185]: Place the comment above this line.

File: src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.test.tsx [hunk 4]
```diff
@@ -113,4 +118,319 @@ describe('Testing Advertisement Entry Component', () => {
...
+    // After clicking the button, the dropdown should be visible
+    expect(queryByText('Edit')).toBeInTheDocument();
+
+    // Click again to close the dropdown
+    fireEvent.click(optionsButton);
+
+    // After the second click, the dropdown should be hidden again
+    expect(queryByText('Edit')).toBeNull();
+  });
+  test('Updates the advertisement and shows success toast on successful update', async () => {
```
Comment (beingnoble03) [inline src/components/Advertisements/core/AdvertisementEntry/AdvertisementEntry.test.tsx:166]: Leave an empty line after every test.

**Review by beingnoble03 — CHANGES_REQUESTED** [review]
@Vaishali054 please address the comments.

File: src/components/Advertisements/Advertisements.test.tsx (hunk 1 above)
Comment (Vaishali054) [reply src/components/Advertisements/Advertisements.test.tsx:166]: yes

**Comment by Vaishali054** [comment]
@beingnoble03 updated the codebase with said changes

**Review by beingnoble03 — APPROVED** [review]
Lgtm. Thanks @Vaishali054!

**Pull request merged**